### ISSUE TITLE
Can we rename HAVE_CONFIG_H?

### ISSUE THREAD

**Issue by mtfurlan** [cpputest/cpputest#1829]
I'm working with [asn1c](https://github.com/mouse07410/asn1c) and CppUTest, and both projects use `HAVE_CONFIG_H` to mean things, and CppUTest defining it as a public compile definition is causing issues for the asn1c code

I would really like if we could rename the one in this repo to `CPPUTEST_HAVE_CONFIG_H`
From the comments in the file, it looks like it isn't supposed to be user facing, but searching github issues shows that I personally have had to define it manually in the past

So can we deprecate `HAVE_CONFIG_H` in favour of `CPPUTEST_HAVE_CONFIG_H` but support both?
```
#if defined(HAVE_CONFIG_H) || defined(CPPUTEST_HAVE_CONFIG_H)
```

**Comment by mtfurlan** [comment]
asn1c is renaming their HAVE_CONFIG_H to HAVE_ASN1C_CONFIG_H: https://github.com/mouse07410/asn1c/pull/202

**Comment by mtfurlan** [comment]
Okay so it turns out it's not easy, `HAVE_CONFIG_H` is coming from [autoconf](https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Configuration-Headers.html)

So I think the fix is to not expose the define publicly in [CMakeLists.txt](https://github.com/cpputest/cpputest/blob/master/CMakeLists.txt#L129-L132), that should be PRIVATE not PUBLIC

**Comment by basvodde** [comment]
Yes, I think I looked at changing the HAVE_CONFIG_H in autoconf and I couldn't find it :( The proposed solution sounds good to me! Probably @thetic has an opinion too.

**Comment by thetic** [comment]
The macro needs to be public. For example, TestHarness.h includes CppUTestConfig.h includes CppUTestGeneratedConfig.h which needs `HAVE_CONFIG_H` to include the generated header.

**Comment by thetic** [comment]
I agree that the generic name is prone to collisions, but I think we would need to convince autoconf to use something else. I'm no expert, but i'll see what i can find in the docs.
